src/mapper.rs

```diff
@@ -153,15 +192,51 @@ impl<'s> ProguardMapper<'s> {
                         .members
                         .entry(obfuscated)
                         .or_insert_with(|| Vec::with_capacity(1));
-                    members.push(MemberMapping {
+
+                    let member_mapping = MemberMapping {
                         startline,
                         endline,
                         original_class,
                         original,
                         original_startline,
                         original_endline,
-                    });
-                }
+                    };
+                    members.push(member_mapping.clone());
+
+                    // If the next line has the same leading line range then this method
+                    // has been inlined by the code minification process, as a result
+                    // it can't show in method traces and can be safely ignored.
+                    if let Some(ProguardRecord::Method {
+                        line_mapping: Some(next_line),
+                        ..
+                    }) = records.peek()
+                    {
+                        if let Some(current_line_mapping) = current_line {
+                            if (current_line_mapping.startline == next_line.startline)
+                                && (current_line_mapping.endline == next_line.endline)
+                            {
+                                continue;
+                            }
+                        }
+                    }
+
+                    let key = format!(
+                        "{}-{}-{}-{}",
+                        class.obfuscated, obfuscated, arguments, original
+                    );
+                    if let std::collections::hash_map::Entry::Vacant(e) = unique_methods.entry(key)
+                    {
```

Comment:
you don’t need to stringly-type this, you can just pass a tuple. And also use a `HashSet` instead, like so:

```suggestion
                    let key = (class.obfuscated, obfuscated, arguments, original);
                    if unique_methods.insert(key) {
```

The `HashSet::insert` will return `true` if the entry was newly inserted, and `false` otherwise.

Reply:
nice! :)